src/main/java/picard/illumina/NewIlluminaBasecallsConverter.java

```diff
@@ -161,10 +161,21 @@ public void doTileProcessing() {
         tileProcessingExecutor.shutdown();
 
         awaitThreadPoolTermination("Reading executor", tileProcessingExecutor);
-        awaitThreadPoolTermination("Tile completion executor", completedWorkExecutor);
 
-        barcodeWriterThreads.values().forEach(ThreadPoolExecutor::shutdown);
-        barcodeWriterThreads.forEach((barcode, executor) -> awaitThreadPoolTermination(barcode + " writer", executor));
+        // if there was an exception reading then purge the completion and writer queues and initiate shutdown.
+        if(tileProcessingExecutor.exception != null) {
+            completedWorkExecutor.purge();
+            barcodeWriterThreads.values().forEach(ThreadPoolExecutor::purge);
+
+            List<Runnable> tasksStillRunning = new ArrayList<>(completedWorkExecutor.shutdownNow());
```

Comment:
Since you don't actually use the `List` except to compute the size it might be simpler to use `size()` here and write `+= executor.shutdownNow().size` in the `forEach()` below.